securedrop_client/logic.py

```diff
@@ -468,6 +471,8 @@ def update_sync(self):
         logged_out = not bool(self.api)
         if logged_out or self.api_paused:
             self.gui.show_sync(self.last_sync())
+        else:
```

Comment:
this entire function should be just one line (`self.gui.show_sync(self.last_sync())`) after you remove the update_sync calls from `login` and `update_sources` so that it is only called when offline

Reply:
`update_sync` should be just the single line. In that case, where's the logic to check if the user is logged in or the queue is paused going to go..? This method is called by a QTimer every 30 seconds so needs to have the check logic in here otherwise the `last_sync` will always be displayed (which is the behaviour we're trying to avoid, right..?). I'm probably missing something here. :-((((

Reply:
I think the code would be easier to a) rename `update_sync` to `show_last_sync_time` because the only time we update the sync timestamp is when a sync is successful, and b) only start calls to `update_sync` in `logout` and `login_in_offline_mode` (and when temporarily switching to offline mode when there is a connection issue that causes the queues to pause, which can be implemented outside of this pr) 

calling `update_sync` on `login` and `update_sources` is confusing to maintain, imo

Reply:
You can call `update_sync` and `self.sync_timer.start()` on `logout` and `login_in_offline_mode`, and stop the timer on `login`. Also this will be clearer if named: `show_last_sync_time` and `self.show_last_timestamp_timer.start()` or something similar, to avoid confusion with our other sync timer for metadata syncs as well as avoid confusion about when the timestamp is getting updated vs just displayed.

Reply:
That'll work. :+1: